stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -169,62 +180,183 @@ public static String getErrorMessage(Throwable err) {
         } else if (throwable instanceof TooLongFrameException) {
-            errorCode = ErrorCode.WsPayloadTooBigError;
+            errorCode = WebSocketConstants.ErrorCode.WsPayloadTooBigError;
         } else if (throwable instanceof CodecException) {
-            errorCode = ErrorCode.WsProtocolError;
+            errorCode = WebSocketConstants.ErrorCode.WsProtocolError;
         } else if (throwable instanceof WebSocketHandshakeException) {
-            errorCode = ErrorCode.WsInvalidHandshakeError;
+            errorCode = WebSocketConstants.ErrorCode.WsInvalidHandshakeError;
         } else if (throwable instanceof IOException) {
-            errorCode = ErrorCode.WsConnectionError;
+            errorCode = WebSocketConstants.ErrorCode.WsConnectionError;
             cause = createErrorCause(throwable.getMessage(), IOConstants.ErrorCode.GenericError.errorCode(),
-                                     IO_PACKAGE_ID);
+                    IOConstants.IO_PACKAGE_ID);
             message = "IO Error";
         }
         return new WebSocketException(errorCode, message, cause);
     }
 
-    public static ErrorValue createErrorCause(String message, String reason, BPackage packageName) {
+    private static ErrorValue createErrorCause(String message, String reason, BPackage packageName) {
 
         MapValue<String, Object> detailRecordType = BallerinaValues.createRecordValue(
                 packageName, WebSocketConstants.WEBSOCKET_ERROR_DETAILS);
         MapValue<String, Object> detailRecord = BallerinaValues.createRecord(detailRecordType, message, null);
         return BallerinaErrors.createError(reason, detailRecord);
     }
 
-    public static MapValue<String, Object> createDetailRecord(String errMsg) {
+    static MapValue<String, Object> createDetailRecord(String errMsg) {
         return createDetailRecord(errMsg, null);
     }
 
-    public static MapValue<String, Object> createDetailRecord(String errMsg, ErrorValue cause) {
+    static MapValue<String, Object> createDetailRecord(String errMsg, ErrorValue cause) {
         MapValue<String, Object> detail = BallerinaValues.createRecordValue(HttpConstants.PROTOCOL_HTTP_PKG_ID,
                                                                             WebSocketConstants.WEBSOCKET_ERROR_DETAILS);
         return BallerinaValues.createRecord(detail, errMsg, cause);
     }
 
+    /**
+     * Check whether the client's config has the `retryConfig` property or not.
+     *
+     * @param webSocketClient - the WebSocket client.
+     * @return If the client's config has the retry config, then return true.
+     */
+    public static boolean hasRetryConfig(ObjectValue webSocketClient) {
+        return webSocketClient.getMapValue(CLIENT_ENDPOINT_CONFIG).
+                getMapValue(WebSocketConstants.RETRY_CONFIG) != null;
+    }
+
+    private static void waitForHandshake(ObjectValue webSocketClient, CountDownLatch countDownLatch) {
```

Comment:
Shall we maintain vertical ordering and move this method after the caller not before it?